### PR TITLE
Show users

### PR DESCRIPTION
**Pull request by kmandziy**
- generated policies
- show users only for admin

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  def index
+    @users = Users::IndexFacade.new.users
```
Comment (volodialeobit) [inline app/controllers/users_controller.rb]: nope
it should be like
```
@facade = Users::IndexFacade.new
..
(on view)
@facade.users
```

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    authorize @users
```
Comment (volodialeobit) [inline app/controllers/users_controller.rb]: move to before_action, and base it on `User`, not `@users`

File: app/policies/application_policy.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  def edit?
+    update?
+  end
+
+  def destroy?
+    false
+  end
+
+  class Scope
```
Comment (volodialeobit) [inline app/policies/application_policy.rb]: you don't really need this class, I believe

File: app/policies/application_policy.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+class ApplicationPolicy
+  attr_reader :user, :record
+
+  def initialize(user, record)
+    @user = user
+    @record = record
+  end
+
+  def index?
```
Comment (volodialeobit) [inline app/policies/application_policy.rb]: I recommend to create a method like this:
```
def not_allowed
  false
end
```
and analogically for true
then..
change here and reuse in child policies like
```
alias_method :index?, :not_allowed
```
etc

File: app/policies/user_policy.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class UserPolicy < ApplicationPolicy
+  def index?
```
Comment (volodialeobit) [inline app/policies/user_policy.rb]: here you can go with `delegate` for `admin?` method and `alias_method` the same way I described for application_policy

File: app/views/users/index.html.haml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+.container
+  .row
+    .col-sm-5.pt-2
+    - @users.each do |user|
```
Comment (volodialeobit) [inline app/views/users/index.html.haml]: `@facade.users` as I mentioned above

**Comment by volodialeobit** [comment]
Kristina, ia nedovolen, gde changes?
@kmandziy

File: app/policies/application_policy.rb [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+  def allowed
+    true
+  end
+
+  def not_allowed
+    false
+  end
+
+  alias index?   allowed
```
Comment (volodialeobit) [inline app/policies/application_policy.rb]: https://blog.bigbinary.com/2012/01/08/alias-vs-alias-method.html
https://stackoverflow.com/a/4763178

File: app/policies/user_policy.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class UserPolicy < ApplicationPolicy
+  delegate :admin?, to: :user
+
+  alias index? admin?
```
Comment (volodialeobit) [inline app/policies/user_policy.rb]: alias_method

File: app/policies/application_policy.rb [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+  def not_allowed
+    false
+  end
+
+  alias_method :index?,   :allowed
+  alias_method :show?,    :allowed
+  alias_method :create?,  :not_allowed
+  alias_method :update?,   :not_allowed
+  alias_method :destroy?,  :not_allowed
```
Comment (oostashchenko-leobit) [inline app/policies/application_policy.rb]: align it somehow, to many spaces and not in line

**Pull request merged**